lib/Parse/ParseExpr.cpp

```diff
@@ -2399,6 +2399,39 @@ parseClosureSignatureIfPresent(SmallVectorImpl<CaptureListEntry> &captureList,
     }
   }
 
+  if (!params)
+    return invalid;
+
+  // If this was a closure declaration (maybe even trailing)
+  // it's imperative to detect any parameter list destructuring
+  // early on, see SE-0110 for more details.
+  for (unsigned i = 0, e = params->size(); i != e; ++i) {
+    auto *param = params->get(i);
+    if (!param->isInvalid())
+      continue;
+
+    auto paramRange = CharSourceRange(
+        SourceMgr, params->getStartLoc(),
+        Lexer::getLocForEndOfToken(SourceMgr, params->getEndLoc()));
```

Comment:
Ah ok, for for example (x, (y, z), (a, b)) is going to result in (x, arg1, arg2) and 2 "let" statements - "let (y, z) = arg1" and "let (a, b) = arg2", WDYT?